### ISSUE TITLE
LND is constantly hanging on "LTND: Waiting for chain backend to finish sync, start_height="

### ISSUE THREAD

**Issue by Perlover** [lightningnetwork/lnd#2195]
### Background

Describe your issue here.

### Your environment
* version of `lnd` - **0.5.0-beta commit=v0.5.1-beta-rc1**
* which operating system (`uname -a` on *Nix) - **CentOS 6.\***
* version of `btcd`, `bitcoind`, or other backend - **bitcoind v0.17.0.1**
* any other relevant environment details

### Steps to reproduce

After normal working of LND i decided stop & start. I made graceful shutdown and after i cannot run node again. I many times stopped bitcoind, lnd - everytime same last log in LND:

    LTND: Waiting for chain backend to finish sync, start_height=

In this time the `lnd` process cunsumes 50-100% cpu.

### Expected behaviour

Norman starting. I have same compiled `lnd`s daemons with same OS & `bitcoind` with `lnd` version. There are everything OK. There are same configs. For example normal behaviour:

    2018-11-17 09:37:17.399 [INF] LTND: Primary chain is set to: bitcoin
    2018-11-17 09:37:17.411 [INF] LNWL: Started listening for bitcoind block notifications via ZMQ on tcp://127.0.0.1:28332
    2018-11-17 09:37:17.411 [INF] LTND: Initializing bitcoind backed fee estimator
    2018-11-17 09:37:17.411 [INF] LNWL: Started listening for bitcoind transaction notifications via ZMQ on tcp://127.0.0.1:28333
    2018-11-17 09:37:19.002 [INF] LNWL: The wallet has been unlocked without a time limit
    2018-11-17 09:37:19.005 [INF] LTND: LightningWallet opened
    2018-11-17 09:37:19.052 [INF] HSWC: Restoring in-memory circuit state from disk
    2018-11-17 09:37:19.082 [INF] HSWC: Payment circuits loaded: num_pending=0, num_open=0
    ...
    2018-11-17 09:37:19.166 [INF] HSWC: Trimming open circuits for chan_id=550338:1779:0, start_htlc_id=0
    2018-11-17 09:37:19.316 [INF] RPCS: RPC server listening on 127.0.0.1:10009
    2018-11-17 09:37:19.316 [INF] RPCS: gRPC proxy started at 127.0.0.1:8080
    2018-11-17 09:37:19.318 [INF] LTND: Waiting for chain backend to finish sync, start_height=550422
    2018-11-17 09:37:19.340 [INF] LNWL: Started rescan from block 0000000000000000000ebf43216644b72c87769086cae234b20e12e0cbca15a4 (height 550422) for 662 addresses
    2018-11-17 09:37:19.348 [INF] LNWL: Starting rescan from block 0000000000000000000ebf43216644b72c87769086cae234b20e12e0cbca15a4
    2018-11-17 09:37:19.351 [INF] LNWL: Rescan finished at 550422 (0000000000000000000ebf43216644b72c87769086cae234b20e12e0cbca15a4)


### Actual behaviour

The last log line same is: `LTND: Waiting for chain backend to finish sync, start_height=NNNN` where is *NNNN* - current last block of bitcoin network and `bitcoind`

    2018-11-17 13:22:51.720 [INF] LTND: Version: 0.5.0-beta commit=v0.5.1-beta-rc1, build=production, logging=default
    2018-11-17 13:22:51.720 [INF] LTND: Active chain: Bitcoin (network=mainnet)
    2018-11-17 13:22:51.720 [INF] CHDB: Checking for schema update: latest_version=6, db_version=6
    2018-11-17 13:22:51.754 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
    2018-11-17 13:22:51.754 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
    2018-11-17 13:22:51.754 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
    2018-11-17 13:23:09.572 [INF] LNWL: Opened wallet
    2018-11-17 13:23:09.699 [INF] LTND: Primary chain is set to: bitcoin
    2018-11-17 13:23:09.711 [INF] LTND: Initializing bitcoind backed fee estimator
    2018-11-17 13:23:09.711 [INF] LNWL: Started listening for bitcoind block notifications via ZMQ on tcp://127.0.0.1:28332
    2018-11-17 13:23:09.711 [INF] LNWL: Started listening for bitcoind transaction notifications via ZMQ on tcp://127.0.0.1:28333
    2018-11-17 13:23:11.353 [INF] LNWL: The wallet has been unlocked without a time limit
    2018-11-17 13:23:11.356 [INF] LTND: LightningWallet opened
    2018-11-17 13:23:11.395 [INF] HSWC: Restoring in-memory circuit state from disk
    2018-11-17 13:23:11.407 [INF] HSWC: Payment circuits loaded: num_pending=0, num_open=0
    2018-11-17 13:23:11.487 [INF] HSWC: Trimming open circuits for chan_id=550343:866:0, start_htlc_id=0
    2018-11-17 13:23:11.487 [INF] HSWC: Trimming open circuits for chan_id=550343:867:0, start_htlc_id=0
    2018-11-17 13:23:11.487 [INF] HSWC: Trimming open circuits for chan_id=550338:1772:0, start_htlc_id=0
    ...
    2018-11-17 13:23:11.490 [INF] HSWC: Trimming open circuits for chan_id=550341:613:0, start_htlc_id=0
    2018-11-17 13:23:11.490 [INF] HSWC: Trimming open circuits for chan_id=550343:871:0, start_htlc_id=0
    2018-11-17 13:23:11.490 [INF] HSWC: Trimming open circuits for chan_id=550346:263:0, start_htlc_id=0
    2018-11-17 13:23:11.490 [INF] HSWC: Trimming open circuits for chan_id=550338:1652:0, start_htlc_id=0
    2018-11-17 13:23:11.515 [INF] RPCS: RPC server listening on 127.0.0.1:10009
    2018-11-17 13:23:11.515 [INF] RPCS: gRPC proxy started at 127.0.0.1:8080
    2018-11-17 13:23:11.517 [INF] LTND: Waiting for chain backend to finish sync, start_height=550441

I wait hours... Nothing. I stop and start again `lnd` (graceful shuttdown) - same problem. I stop lnd + bitcoind and run its again - same problem. I stop lnd and wait 1-2 hours and start again - same problem.

In this time the `top` shows to me like this:
    PID USER      PR  NI  VIRT  RES  SHR S %CPU %MEM    TIME+  COMMAND                                                                                                                                                                                                                                                          
   3489 bitcoin   20   0 3238m  68m  11m S 105.5  0.1  15:36.09 lnd        

My LND config is:

    [Application Options]
    rpclisten=localhost:10009
    restlisten=localhost:8080
    debuglevel=info
    debughtlc=false
    autopilot.active=1
    autopilot.maxchannels=1024
    #autopilot.maxchansize=9000000
    autopilot.allocation=1.0
    
    # You need configure router for port forwarding
    externalip=MY_IP:9735
    listen=MY_IP:9735
    alias=MY_ALIAS
    
    datadir=/home/bitcoin/.lnd/data-mainnet/
    adminmacaroonpath=/home/bitcoin/.lnd/admin-mainnet.macaroon
    readonlymacaroonpath=/home/bitcoin/.lnd/readonly-mainnet.macaroon
    invoicemacaroonpath=/home/bitcoin/.lnd/invoice-mainnet.macaroon
    
    [Bitcoin]
    bitcoin.mainnet=1
    bitcoin.active=1
    bitcoin.node=bitcoind
    bitcoin.feerate=100
    bitcoin.basefee=1000
    
    [Bitcoind]
    bitcoind.rpcuser=lnd
    bitcoind.rpcpass=MY_PASS_HERE
    bitcoind.zmqpubrawblock=tcp://127.0.0.1:28332
    bitcoind.zmqpubrawtx=tcp://127.0.0.1:28333

The `bitcoind` config is:

    txindex=1
    server=1
    daemon=1
    
    pid=/home/bitcoin/.mainnet-bitcoind.pid
    
    # Will beother upnp support - through 'upnpc' utility
    upnp=0
    
    # ZMQ for lnd...
    zmqpubrawblock=tcp://127.0.0.1:28332
    zmqpubrawtx=tcp://127.0.0.1:28333
    
    # Auth key for lnd
    rpcauth=lnd:MY_KEY_HERE
    
    # You need configure router for port forwarding
    externalip=MY_IP_HERE

`mainnet-lncli getinfo` shows to me now:

    {
	"identity_pubkey": "MY_NODE_ID",
	"alias": "MY_ALIAS",
	"num_pending_channels": 0,
	"num_active_channels": 0,
	"num_peers": 0,
	"block_height": 550444,
	"block_hash": "0000000000000000000c56c6fc713489787ab6c53e572484d71516c3865f16b8",
	"synced_to_chain": false,
	"testnet": false,
	"chains": [
	    "bitcoin"
	],
	"uris": [
	    "MY_NODE_ID@MY_IP:9735"
	],
	"best_header_timestamp": "0",
	"version": "0.5.0-beta commit=v0.5.1-beta-rc1",
	"num_inactive_channels": 59
    }

P.S. I do regular incremental backups each minute by `rsync` to other machine so i have all binary files before last normal working. So I have a  option to copy all the files, but I want to understand the problem.

**Comment by Perlover** [comment]
I don't think that problem with bitcoind because before the last fatal stopping i had `bitcoind` worked and i don't stopped it. I changed only option to `autopilot.allocation=1.0` of lnd and started lnd again. The logs of last shutdown before problem:

    2018-11-17 09:25:58.712 [INF] CRTR: Examining Channel Graph for zombie channels
    2018-11-17 09:25:58.943 [INF] CRTR: Pruning 34 Zombie Channels
    2018-11-17 09:26:23.822 [INF] LTND: Received shutdown request.
    2018-11-17 09:26:23.822 [INF] LTND: Shutting down...
    2018-11-17 09:26:23.822 [INF] LTND: Gracefully shutting down.
    2018-11-17 09:26:23.822 [INF] ATPL: Autopilot Agent stopping
    2018-11-17 09:26:23.822 [INF] HSWC: Block epoch canceled, decaying hash log shutting down
    2018-11-17 09:26:23.822 [INF] HSWC: Removing channel link with ChannelID(XXXXXXXXXXXXX)
    ...
    2018-11-17 09:26:23.823 [INF] CRTR: Channel Router shutting down
    ..
    2018-11-17 09:26:23.823 [INF] CRTR: FilteredChainView stopping
    ..
    2018-11-17 09:26:23.823 [INF] HSWC: HTLC Switch shutting down
    ..
    2018-11-17 09:26:23.824 [INF] HSWC: Removing channel link with ChannelID(XXXXXXXXXXXXXXXXXXX)
    2018-11-17 09:26:23.824 [INF] HSWC: ChannelLink(XXXXXXXXXXXXXXXXXXXXXX:0) is stopping
    ...
    2018-11-17 09:26:23.927 [INF] HSWC: ChannelLink(XXXXXXXXXXXXXXXXXXXXXXXXX:0) has exited
    2018-11-17 09:26:23.928 [INF] UTXN: UTXO nursery shutting down
    2018-11-17 09:26:23.928 [INF] BRAR: Breach arbiter shutting down
    2018-11-17 09:26:23.928 [INF] DISC: Authenticated Gossiper is stopping
    2018-11-17 09:26:23.928 [INF] CNCT: Stopping ChainArbitrator
    2018-11-17 09:26:23.928 [INF] FNDG: Funding manager shutting down
    ...
    2018-11-17 09:26:24.232 [INF] LTND: Shutdown complete
    >> ------ HERE LND was stopped than i started it again
    2018-11-17 09:27:00.977 [INF] LTND: Version: 0.5.0-beta commit=v0.5.1-beta-rc1, build=production, logging=default
    2018-11-17 09:27:00.977 [INF] LTND: Active chain: Bitcoin (network=mainnet)
    2018-11-17 09:27:00.977 [INF] CHDB: Checking for schema update: latest_version=6, db_version=6
    >>> ------ HERE LINES already printed in my post here above. To see there...

**Comment by vegardengen** [comment]
I have the same issue.

How is go dependencies resolved? I had an older go/src/github.com/btcsuite/btcwallet that I'd used to compile dropwtxmgr earlier....could that have been selected?

Deleted that and recomped, restarted - rescanning now. If that theory is correct, the restart that will be without the rescan.

Will report back in a couple of hours, but in the meantime someone that knows might answer you :)

**Comment by Perlover** [comment]
I saw now sources
I see problem here

https://github.com/lightningnetwork/lnd/blob/master/lnd.go#L389

    for {
	    if !signal.Alive() {
		    return nil
	    }
    
	    synced, _, err := activeChainControl.wallet.IsSynced()
	    if err != nil {
		    return err
	    }
    
	    if synced {
		    break
	    }
    
	    time.Sleep(time.Second * 1)
    }

The daemon is spinning in this cycle - this explains the CPU consumption...

**Comment by vegardengen** [comment]
This issue was there, but was supposedly fixed by this commit. This is why I am questioning dependency resolving...

commit 6e06a617b22a82ec3c2bed1fe86160dc3f1db15f
Author: Olaoluwa Osuntokun <laolu32@gmail.com>
Date:   Thu Nov 15 19:08:44 2018 -0800

    build: update to latest version of btcwallet
    
    In this commit, we update to the latest version of btcwallet. This
    version includes a bug fix which ensures that the wallet birthday sanity
    check only executes once and not each time the deamon is restarted.

**Comment by Perlover** [comment]
@vegardengen Thanks! Right now i will try to get fresh commit and will recompile the lnd
I will let know result

**Comment by Perlover** [comment]
I created new issue 

https://github.com/lightningnetwork/lnd/issues/2196

Now i cannot compile from fresh commit. So right now i cannot check resolving of https://github.com/lightningnetwork/lnd/commit/6e06a617b22a82ec3c2bed1fe86160dc3f1db15f

**Comment by Perlover** [comment]
I recomiled to commit `4da1c867c3209dab4e4a824b73d89fc38b616b37` (tag `v0.5.1-beta-rc2`), installed and run again. Same problem, last logs:

    2018-11-17 15:09:38.482 [INF] LTND: Version: 0.5.1-beta commit=v0.5.1-beta-rc2, build=production, logging=default
    2018-11-17 15:09:38.482 [INF] LTND: Active chain: Bitcoin (network=mainnet)
    2018-11-17 15:09:38.482 [INF] CHDB: Checking for schema update: latest_version=6, db_version=6
    2018-11-17 15:09:38.515 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
    2018-11-17 15:09:38.516 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
    2018-11-17 15:09:38.516 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
    2018-11-17 15:09:49.342 [INF] LNWL: Opened wallet
    2018-11-17 15:09:49.463 [INF] LTND: Primary chain is set to: bitcoin
    2018-11-17 15:09:49.482 [INF] LTND: Initializing bitcoind backed fee estimator
    2018-11-17 15:09:49.482 [INF] LNWL: Started listening for bitcoind block notifications via ZMQ on tcp://127.0.0.1:28332
    2018-11-17 15:09:49.482 [INF] LNWL: Started listening for bitcoind transaction notifications via ZMQ on tcp://127.0.0.1:28333
    2018-11-17 15:09:51.163 [INF] LNWL: The wallet has been unlocked without a time limit
    2018-11-17 15:09:51.166 [INF] LTND: LightningWallet opened
    2018-11-17 15:09:51.195 [INF] HSWC: Restoring in-memory circuit state from disk
    2018-11-17 15:09:51.201 [INF] HSWC: Payment circuits loaded: num_pending=0, num_open=0
    ...
    2018-11-17 15:09:51.321 [INF] RPCS: RPC server listening on 127.0.0.1:10009
    2018-11-17 15:09:51.321 [INF] RPCS: gRPC proxy started at 127.0.0.1:8080
    2018-11-17 15:09:51.323 [INF] LTND: Waiting for chain backend to finish sync, start_height=550452

**Comment by Perlover** [comment]
I saw sources and i think problem may be there:

https://github.com/lightningnetwork/lnd/blob/master/lnwallet/btcwallet/btcwallet.go#L716

**Comment by Roasbeef** [comment]
It's the wallet catching up block hashes from the birthday to current time. If you run with the debug level of `LNWL=trace`, you'll see the progress. Older wallets will take longer to complete this process. If you're using remote node, then it'll take longer due to the additional round trip time.

**Comment by Perlover** [comment]
I had run some nodes at one time. All nodes were synchronised. And i only stopped and run again. Other nodes started this not. Where are older wallets? It's bug! Now i am not computer,  i will try your debug level but it will not help to me...

**Comment by Perlover** [comment]
Here is second server with same problem:

    2018-11-18 13:29:10.989 [INF] HSWC: Trimming open circuits for chan_id=550343:871:0, start_htlc_id=0
    2018-11-18 13:29:10.989 [INF] HSWC: Trimming open circuits for chan_id=550346:263:0, start_htlc_id=0
    2018-11-18 13:29:10.989 [INF] HSWC: Trimming open circuits for chan_id=550338:1652:0, start_htlc_id=0
    2018-11-18 13:29:11.015 [INF] RPCS: RPC server listening on 127.0.0.1:10009
    2018-11-18 13:29:11.015 [INF] RPCS: gRPC proxy started at 127.0.0.1:8080
    2018-11-18 13:29:11.017 [INF] LTND: Waiting for chain backend to finish sync, start_height=550590
    2018-11-18 13:29:52.955 [INF] RPCS: [debuglevel] changing debug level to: LNWL=trace
    2018-11-18 13:29:52.958 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.960 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.961 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.963 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.965 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.966 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.968 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.969 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.971 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.973 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.974 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.975 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.977 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.978 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.980 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.981 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.983 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.984 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.986 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.987 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.989 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.990 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.991 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.993 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.994 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.996 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.997 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:52.999 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:53.000 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:53.002 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:53.003 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:53.004 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:53.006 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:53.007 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:53.008 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:53.010 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:53.011 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:53.013 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    ...
    2018-11-18 13:29:59.252 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.253 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.254 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.256 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.257 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.259 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.260 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.262 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.263 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.265 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.267 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.268 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.270 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.271 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.273 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.274 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.276 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.277 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.279 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.280 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.282 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.283 [DBG] LNWL: Checking next birthday block candidate: height=550326, hash=00000000000000000007a6e7540eb795f8343825c8d153e0ac4ea66c1aad49cc, timestamp=2018-11-15 12:10:38 +0100 CET
    2018-11-18 13:29:59.284 [INF] RPCS: [debuglevel] changing debug level to: LNWL=info

**Comment by vegardengen** [comment]
@mariodian In my case, it also happened on every restart. Turned out I had another build of btcwallet in my GO source repository, sitting at go/src/github.com/btcsuite/btcwallet

I deleted this, then make clean and built LND again - and then it rescanned once more, but stopped after that.

**Comment by mariodian** [comment]
@vegardengen thanks! In my case it was like this:

I was on RC1 commit and it was rescanning every time I restarted. With the latest commit I first `make clean` and then `make & make install`, rescanned once more but after another restart it's all good.

I suppose btcwallet from that repo was built too but `make clean` on the lnd repo and then rebuilding it fixed that?

**Comment by Perlover** [comment]
Wallets are encrypted and seeds are encrypted too

**Comment by wpaulino** [comment]
@Perlover can you try running the patch provided by https://github.com/lightningnetwork/lnd/pull/2200 locally?

To do so, follow these steps:
1. `cd $GOPATH/src/github.com/lightningnetwork/lnd`
2. `git fetch origin pull/2200/head:birthday-block-infinite-loop`
3. `git checkout birthday-block-infinite-loop`
4. `make clean && make && make install`

**Comment by Perlover** [comment]
I will try rigth now patch, please wait

**Comment by Perlover** [comment]
Now it's progress!

    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550333:1398:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550338:1770:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550338:1501:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550333:1401:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550350:2064:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550338:1768:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550338:1774:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550361:390:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550333:1397:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550338:1150:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550364:1129:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550346:267:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550341:613:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550343:871:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550346:263:0, start_htlc_id=0
    2018-11-19 09:40:23.097 [INF] HSWC: Trimming open circuits for chan_id=550338:1652:0, start_htlc_id=0
    2018-11-19 09:40:23.128 [INF] RPCS: RPC server listening on 127.0.0.1:10009
    2018-11-19 09:40:23.128 [INF] RPCS: gRPC proxy started at 127.0.0.1:8080
    2018-11-19 09:40:23.130 [INF] LTND: Waiting for chain backend to finish sync, start_height=550692
    2018-11-19 09:40:23.477 [INF] LNWL: Started rescan from block 00000000000000000007fc22d66b6b9d67d30365b3a84fa2660c47e45d14812b (height 549948) for 747 addresses
    2018-11-19 09:40:23.486 [INF] LNWL: Starting rescan from block 00000000000000000007fc22d66b6b9d67d30365b3a84fa2660c47e45d14812b
    2018-11-19 09:40:23.497 [INF] LNWL: Catching up block hashes to height 549953, this might take a while
    2018-11-19 09:40:23.699 [INF] LNWL: Done catching up block hashes
    2018-11-19 09:40:23.699 [INF] LNWL: Rescanned through block 00000000000000000005909a80526d1297c883abb08f3ee5cbd2a98c754c37dc (height 549953)
    2018-11-19 09:40:38.339 [INF] LNWL: Catching up block hashes to height 550000, this might take a while
    2018-11-19 09:40:38.345 [INF] LNWL: Done catching up block hashes
    2018-11-19 09:40:38.345 [INF] LNWL: Rescanned through block 000000000000000000223b7a2298fb1c6c75fb0efc28a4c56853ff4112ec6bc9 (height 550000)

**Comment by Perlover** [comment]
The node was started. The process is continueing now but now i think everything will work! I see now the logs of channels! Thanks!

**Comment by Perlover** [comment]
The second node with same problem was started too! :)
Its logs (tail)

    2018-11-19 09:47:39.863 [INF] RPCS: gRPC proxy started at 127.0.0.1:8080
    2018-11-19 09:47:39.864 [INF] LTND: Waiting for chain backend to finish sync, start_height=550692
    2018-11-19 09:47:39.934 [INF] LNWL: Started rescan from block 00000000000000000010d2c2f121984004e1f83e1da7abed7e30ceb4258d8ba0 (height 549950) for 662 addresses
    2018-11-19 09:47:39.942 [INF] LNWL: Starting rescan from block 00000000000000000010d2c2f121984004e1f83e1da7abed7e30ceb4258d8ba0
    2018-11-19 09:47:39.952 [INF] LNWL: Catching up block hashes to height 549953, this might take a while
    2018-11-19 09:47:40.074 [INF] LNWL: Done catching up block hashes
    2018-11-19 09:47:40.074 [INF] LNWL: Rescanned through block 00000000000000000005909a80526d1297c883abb08f3ee5cbd2a98c754c37dc (height 549953)
    2018-11-19 09:47:53.779 [INF] LNWL: Catching up block hashes to height 550000, this might take a while
    2018-11-19 09:47:53.782 [INF] LNWL: Done catching up block hashes
    2018-11-19 09:47:53.782 [INF] LNWL: Rescanned through block 000000000000000000223b7a2298fb1c6c75fb0efc28a4c56853ff4112ec6bc9 (height 550000)

**Comment by Roasbeef** [comment]
Excellent! We'll close out this issue once that patch is merged in.
